src/taskpane/server.js

```diff
@@ -1511,53 +1788,83 @@ function hideColumn(sheet, columnNumber, rowLength, bgColor) {
+    var fields = model.fields;
+    var columnCount = Object.keys(fields).length + 1;
+    var rowCount = currentSheet.getLastRow();
+    if (rowCount == 0) { rowCount = 1; } //avoid errors
+
+    //reset each column color schema (depending on property type)
+    for (var j = 1; j < columnCount; j++) {
+      var isHidden = fields[j - 1].isHidden;
+      if (!isHidden) {
+        var subColumnRange = currentSheet.getRange(2, j, rowCount, 1);
+        var fieldType = fields[j - 1].type;
+        var isReadOnly = fields[j - 1].isReadOnly;
+
+        if (fieldType == fieldTypeEnums.id || fieldType == fieldTypeEnums.subId || isReadOnly) {
+          subColumnRange.setBackground(model.colors.bgReadOnly);
+          subColumnRange.clearNote();
+        }
+        else {
+          subColumnRange.setBackground(model.colors.header);
+          subColumnRange.clearNote();
+        }
       }
     }
+  }
+  else {
+    Excel.run(function (ctx) {
+      var fields = model.fields;
+      var columnCount = Object.keys(fields).length;
 
-    //complete data range from old data
-    var sheet = ctx.workbook.worksheets.getActiveWorksheet();
-    var range = sheet.getRangeByIndexes(1, 0, rowCount + 1, columnCount);
+      //get the previous data number of rows
+      var rowCount;
+      var sheetOldData = sheetRangeOld.values;
 
-    //reset each column color schema (depending on property type)
-    for (var j = 0; j < columnCount; j++) {
+      for (var i = 0; i < sheetOldData.length; i++) {
+        // stop at the first row that is fully blank
+        if (sheetOldData[i].join("") === "") {
+          break;
+        }
+        else {
+          rowCount = i;
+        }
+      }
 
-      var subColumnRange = range.getColumn(j);
-      var fieldType = fields[j].type;
-      var isReadOnly = fields[j].isReadOnly;
-      var bgColor;
+      //complete data range from old data
+      var sheet = ctx.workbook.worksheets.getActiveWorksheet();
+      var range = sheet.getRangeByIndexes(1, 0, rowCount + 1, columnCount);
 
-      if (fieldType == fieldTypeEnums.id || fieldType == fieldTypeEnums.subId || isReadOnly) {
+      //reset each column color schema (depending on property type)
+      for (var j = 0; j < columnCount; j++) {
 
-        subColumnRange.format.fill.color = model.colors.bgReadOnly;
-      }
-      else {
+        var subColumnRange = range.getColumn(j);
+        var fieldType = fields[j].type;
+        var isReadOnly = fields[j].isReadOnly;
+        var bgColor;
+
+        if (fieldType == fieldTypeEnums.id || fieldType == fieldTypeEnums.subId || isReadOnly) {
+
+          subColumnRange.format.fill.color = model.colors.bgReadOnly;
+        }
+        else {
+
+          subColumnRange.format.fill.clear();
+        }
 
-        subColumnRange.format.fill.clear();
       }
+      range.delete(Excel.DeleteShiftDirection.up);
+      return ctx.sync();
+    }).catch(function (error) {
+      if (error instanceof OfficeExtension.Error) {

```

Comment:
Looks like this if block does nothing? We area always failing quietly? If so can you add a comment?

Reply:
Yes, we want it to fail quietly. I added a comment.